napari/add_layers.py

```diff
@@ -0,0 +1,633 @@
+        Colormaps to use for luminance images. If a string must be the name
+        of a supported colormap from vispy or matplotlib. If a tuple the
+        first value must be a string to assign as a name to a colormap and
+        the second item must be a Colormap. If a dict the key must be a
+        string to assign as a name to a colormap and the value must be a
+        Colormap. If a list then must be same length as the axis that is
+        being expanded and then each colormap is applied to each image.
+    contrast_limits : list (2,)
+        Color limits to be used for determining the colormap bounds for
+        luminance images. If not passed is calculated as the min and max of
+        the image. If list of lists then must be same length as the axis
+        that is being expanded and then each colormap is applied to each
+        image.
+    interpolation : str
+        Interpolation mode used by vispy. Must be one of our supported
+        modes.
+    name : str
+        Name of the layer.
+    metadata : dict
+        Layer metadata.
+    scale : tuple of float
+        Scale factors for the layer.
+    translate : tuple of float
+        Translation values for the layer.
+    opacity : float
+        Opacity of the layer visual, between 0.0 and 1.0.
+    blending : str
+        One of a list of preset blending modes that determines how RGB and
+        alpha values of the layer visual get mixed. Allowed values are
+        {'opaque', 'translucent', and 'additive'}.
+    visible : bool
+        Whether the layer visual is currently being displayed.
+
+    Returns
+    -------
+    viewer : :class:`napari.Viewer`
+        The newly-created viewer.
+    """
+    viewer = Viewer()
+
+    n_images = data.shape[channel]
+
+    name = ensure_iterable(name)
+
+    base_colormaps = ['cyan', 'yellow', 'magenta', 'red', 'green', 'blue']
+    if colormap is None:
+        colormap = itertools.cycle(base_colormaps)
+    else:
+        colormap = ensure_iterable(colormap)
+
+    # If one pair of clim values is passed then need to iterate them to
+    # all layers.
+    if contrast_limits is not None and not is_iterable(contrast_limits[0]):
+        contrast_limits = itertools.repeat(contrast_limits)
+    else:
+        contrast_limits = ensure_iterable(contrast_limits)
+
+    zipped_args = zip(range(n_images), colormap, contrast_limits, name)
+    for i, cmap, clims, name in zipped_args:
+        image = data.take(i, axis=channel)
+        layer = viewer.add_image(
+            image,
+            rgb=False,
+            colormap=cmap,
+            contrast_limits=clims,
+            interpolation=interpolation,
+            rendering=rendering,
+            name=name,
+            metadata=metadata,
+            scale=scale,
+            translate=translate,
+            opacity=opacity,
+            blending=blending,
+            visible=visible,
+        )
+    return viewer
+
+
+def view_pyramid(data, *args, **kwargs):
```

Comment:
For the similar reasons that `import *` are discouraged, `*args` and `**kwargs` are also best avoided, especially when the parameter list is itself very clear (see the doc).

Reply:
And this is more true still given that the rest of the file does include all the parameters!

Reply:
this is going to get refactored when we merge pyramid into image so i'll ignore for now